File: 01 - Classical/07 - String matching/01- Naive/string_matching_naive.py
# -*- coding: utf-8 -*-
"""
Created on Sun Feb 13 22:53:36 2022

@author: AVERKHO
"""

def string_matching_naive(text,pattern):
    
        
    m=len(text)
    n=len(pattern)
    
    found=False
    
    for i in range(m-n+1):
        
        j=0
        
        while(j<n):
            
            if text[i+j]!=pattern[j]:
                
                break
            
            if j==n-1:
                
                print('pattern occurs at shift {}'.format(i))
                found=True
            
            j+=1
        
    if not(found):
        
        print("Pattern '{}' was not found in string '{}'".format(pattern,text))
    


if __name__=="__main__":
    
    text='acsdfegwegaabaabaab'
    pattern='aab'
    
    string_matching_naive(text,pattern)

File: 01 - Classical/07 - String matching/01- Naive/string_matching_naive.cpp
#include <iostream>
#include <string>

void naive_search(std::string text, std::string pattern){

    size_t n=text.size();
    size_t m=pattern.size();

    bool match=false;

    for (int i=0; i<n; ++i){

        int j=0;

        while (j<(n-m+1)){

            if (text[i+j]!=pattern[j]){
               
                break;
            }
            ++j;
        }

        if (j==m){

            std::cout << " The pattern match at shift " << i << std::endl;
            match=true;

        }
    }

    if (match==false){
        std::cout << "No match for " << pattern << " is found in " << text << std::endl;
    }

}

int main(){

    std::string text;
    std::string pattern;

    std::cout << "Enter the string " << std::endl;
    std::cin >> text;

    std::cout << std::endl <<  "Enter the pattern to be searched " << std::endl;
    std::cin >> pattern;

    std::cout << std::endl;

    naive_search(text, pattern);

    return 0;
}